src/fault_injector/fault_types/train_prio_fault.py

```diff
@@ -39,6 +40,12 @@ def resolve_fault(self, tick: int):
         :param tick: the simulation tick in which resolve_fault was called
         :type tick: Integer
         """
-        self.train.train_type.priority = self.old_prio
-        self.interlocking.insert_train_priority_changed(self.train)
-        self.logger.resolve_train_prio_fault(tick, self.configuration.id)
+        if self.train is None:
+            raise ValueError("Fault not injected")
+
+        if self.train is self.get_train(
+            self.simulation_object_updater, self.train.identifier
+        ):
+            self.train.train_type.priority = self.old_prio
+            self.interlocking.insert_train_priority_changed(self.train)
+            self.logger.resolve_train_prio_fault(tick, self.configuration.id)
```

Comment:
Maybe this should be outside of the if clause, so that for every fault injected log there is a fault resolved log.